libs/blocks/aside/aside.js

```diff
@@ -46,32 +47,101 @@ function decorateStaticLinks(el) {
+    <source src="${link.href}" type="video/mp4" />
+  </video>`;
+  container.classList.add('has-video');
+}
+
+function decorateBlockBg(block, node) {
+  const viewports = ['mobile-only', 'tablet-only', 'desktop-only'];
+  const childCount = node.childElementCount;
+  const { children } = node;
+  node.classList.add('background');
+  if (childCount === 2) {
+    children[0].classList.add(viewports[0], viewports[1]);
+    children[1].classList.add(viewports[2]);
+  }
+  [...children].forEach(async (child, index) => {
+    if (childCount === 3) {
+      child.classList.add(viewports[index]);
+    }
+    decorateVideo(child);
+  });
+  if (!node.querySelector(':scope img') && !node.querySelector(':scope video')) {
+    block.style.background = node.textContent;
+    node.remove();
+  }
+}
+
 function decorateLayout(el) {
   const elems = el.querySelectorAll(':scope > div');
   if (elems.length > 1) decorateBlockBg(el, elems[0]);
   const foreground = elems[elems.length - 1];
   foreground.classList.add('foreground', 'container');
+  if (el.classList.contains('split')) decorateMedia(el);
   const text = foreground.querySelector('h1, h2, h3, h4, h5, h6, p')?.closest('div');
   text?.classList.add('text');
   const media = foreground.querySelector(':scope > div:not([class])');
   if (!el.classList.contains('notification')) media?.classList.add('image');
-  const picture = text?.querySelector('picture');
+  const picture = text?.querySelector('p picture');
   const iconArea = picture ? (picture.closest('p') || createTag('p', null, picture)) : null;
   iconArea?.classList.add('icon-area');
   const foregroundImage = foreground.querySelector(':scope > div:not(.text) img')?.closest('div');
   const bgImage = el.querySelector(':scope > div:not(.text) img')?.closest('div');
+  const foregroundMedia = foreground.querySelector(':scope > div:not(.text) video')?.closest('div');
+  const bgMedia = el.querySelector(':scope > div:not(.text) video')?.closest('div');
   const image = foregroundImage ?? bgImage;
-  if (image && !image.classList.contains('text')) {
+  const asideMedia = foregroundMedia ?? bgMedia ?? image;
+  if (asideMedia && !asideMedia.classList.contains('text')) {
     const isSplit = el.classList.contains('split');
-    image.classList.add(`${isSplit ? 'split-' : ''}image`);
+    asideMedia.classList.add(`${isSplit ? 'split-' : ''}image`);
     if (isSplit) {
-      const position = Array.from(image.parentNode.children).indexOf(image);
+      const position = Array.from(asideMedia.parentNode.children).indexOf(asideMedia);
       el.classList.add(`split${!position ? '-right' : '-left'}`);
-      foreground.parentElement.appendChild(image);
+      foreground.parentElement.appendChild(asideMedia);
     }
   } else if (!iconArea) {
     foreground?.classList.add('no-image');
   }
+  if (el.classList.contains('split')
+      && (el.classList.contains('medium')
+      || el.classList.contains('large'))) {
```

Comment:
nit
```suggestion
  if (el.classList.contains('split')
      && (el.classList.contains('medium') || el.classList.contains('large'))) {
```

Reply:
@chrischrischris I have made the requested changes.